src/main/kotlin/eu/europa/ec/eudi/verifier/endpoint/port/input/PostZkpJwkRequest.kt

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright (c) 2023 European Commission
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package eu.europa.ec.eudi.verifier.endpoint.port.input
+
+import arrow.core.raise.Raise
+import eu.europa.ec.eudi.verifier.endpoint.domain.Presentation
+import eu.europa.ec.eudi.verifier.endpoint.domain.RequestId
+import eu.europa.ec.eudi.verifier.endpoint.port.out.persistence.LoadPresentationByRequestId
+import eu.europa.ec.eudi.verifier.endpoint.port.out.persistence.StorePresentation
+import org.springframework.web.reactive.function.server.ServerRequest
+import org.springframework.web.reactive.function.server.awaitBody
+import software.tice.ChallengeRequestData
+import software.tice.ZKPVerifier
+import java.security.KeyFactory
+import java.security.interfaces.ECPublicKey
+import java.security.spec.X509EncodedKeySpec
+import java.util.*
+
+sealed interface ZkpJwkError {
+    data class ProcessingError(val message: String, val error: Throwable) : ZkpJwkError
+}
+
+data class ChallengeRequest(
+    val id: String,
+    val digest: String,
+    val r: String,
+    val proofType: String,
+)
+
+data class EphemeralKeyResponse(
+    val id: String,
+    val kid: String,
+    val kty: String,
+    val crv: String,
+    val x: String,
+    val y: String,
+)
+
+private const val publicKeyPEM = """
```

Comment:
As this is the issuer's public key and we are dealing with multiple key pairs here could you please use a more specific naming? Like `issuerPublicKeyPEM`.